src/mz-debug/src/utils.rs

```diff
@@ -77,43 +79,97 @@ pub async fn get_k8s_auth_mode(
     mz_username: Option<String>,
     mz_password: Option<String>,
     k8s_client: &Client,
-    k8s_namespaces: &Vec<String>,
+    k8s_namespace: &String,
+    mz_instance_name: &String,
 ) -> Result<AuthMode, anyhow::Error> {
-    for namespace in k8s_namespaces.iter() {
-        let materialize_api = Api::<Materialize>::namespaced(k8s_client.clone(), namespace);
-        let object_list = materialize_api
-            .list(&ListParams::default())
-            .await
-            .with_context(|| format!("Failed to get Materialize CR in namespace: {}", namespace))?;
-
-        if !object_list.items.is_empty() {
-            let materialize_cr = &object_list.items[0];
-            let authenticator_kind = materialize_cr.spec.authenticator_kind;
-
-            match authenticator_kind {
-                AuthenticatorKind::None => return Ok(AuthMode::None),
-                AuthenticatorKind::Password => {
-                    if let (Some(mz_username), Some(mz_password)) = (&mz_username, &mz_password) {
-                        return Ok(AuthMode::Password(PasswordAuthCredentials {
-                            username: mz_username.clone(),
-                            password: mz_password.clone(),
-                        }));
-                    } else {
-                        return Err(anyhow::anyhow!(
-                            "mz_username and mz_password are required for password authentication"
-                        ));
-                    }
-                }
-                _ => {
-                    return Err(anyhow::anyhow!(
-                        "Unsupported authenticator kind: {:?}",
-                        authenticator_kind
-                    ));
-                }
+    let materialize_api = Api::<Materialize>::namespaced(k8s_client.clone(), k8s_namespace);
+    let object_list = materialize_api
+        .list(&ListParams::default())
+        .await
+        .with_context(|| {
+            format!(
+                "Failed to get Materialize CR in namespace: {}",
+                k8s_namespace
+            )
+        })?;
+
+    let materialize_cr = object_list
+        .items
+        .into_iter()
+        .find(|item| item.metadata.name.as_ref() == Some(mz_instance_name))
+        .with_context(|| {
+            format!(
+                "Could not find Materialize CR with name: {}",
+                mz_instance_name
+            )
+        })?;
```

Comment:
Why not just use a get? There can only be one with a specific namespace and name.

Reply:
Good call. Changed!